### PR TITLE
[FEAT] Run danger within the scheduler workflow

### PR DESCRIPTION
**Pull request by valerio-castelli**
## Description

This PR improves the execution of PR checks on forked PRs. 

In the previous implementation of the scheduler system, PR checks on forked PRs were executed by identifying _unauthorised_ workflows and scheduling a re-run with the necessary settings to run Danger. However, due to limitations of CircleCI, such workflows may not be unauthorised in the first place, or may not have access to the necessary settings even after they're re-scheduled. This may result in Danger checks not being run on certain forked PRs, which limits our ability to maintain a consistent coding style and give constructive feedbacks to the community.

To solve this issue, we propose a different approach in which Danger is executed directly by the scheduler workflow for all forked PRs, regardless of the authorisation status of the original PR checks workflow. This allows us to improve our PR review process and to offer more consistent suggestions.

This release also carries improvements in the way safety checks are performed and communicated, and introduces additional checks on the integrity of the scheduler process.

## Checklist

- [x] I have followed the indications in the [CONTRIBUTING](../CONTRIBUTING.md).
- [x] The documentation related to the proposed change has been updated accordingly (plus comments in code).
- [ ] I have written new tests for my core changes, as applicable.
- [ ] I have successfully run tests with my changes locally.
- [x] It is ready for review! :rocket:

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -71,13 +72,22 @@ To leverage the scheduler logic in Immuni's iOS and Android applications, the fo
 jobs:
   scheduler:
     docker:
-      - image: cimg/python:3.7.7
+      # The scheduler needs Python, Node.js, and the Danger dependencies
+      # These may be present in the boostrap image or installed manually
+      # The actual configuration varies for each repository
+      - image: cimg/python:3.7.7-node
```
Comment (a-riva) [inline README.md:78]: We are using `circleci/python:3.7-node` now

File: README.md [hunk 2]
```diff
@@ -127,8 +137,11 @@ workflows:
...
-- **GITHUB\_TOKEN**. This is a GitHub API token with read permissions on the repository that must be checked by the scheduler. In Immuni's repos, this is provided by the scheduler CircleCI context. 
-- **REPOSITORY**. This is the repository that must be checked by the scheduler, including the name of the organisation within which said repository is located. In Immuni's repos, this is provided by the Run scheduler step of the scheduler job.
+- **CIRCLE\_WORKFLOW\_ID.** This is the unique identifier of the current execution of the scheduler workflow, and is injected automatically by CircleCI when the scheduler is run. It is used to determine the upper bound of the pipelines to verify. This variable can be left unassigned when running on a local machine, in which case no upper bound is enforced and any pipeline submitted after the execution of a previous scheduler workflow will be verified.
+- **GITHUB\_TOKEN.** This is a GitHub API token with read permissions on the repository that must be checked by the scheduler. In Immuni's repos, this is provided by the scheduler CircleCI context.
```
Comment (a-riva) [inline README.md]:
```suggestion
- **GITHUB\_TOKEN.** This is a GitHub API token with read permissions on the repository that must be checked by the scheduler. In Immuni's repos, this is provided by the *scheduler* CircleCI context. 
```

**Review by a-riva — APPROVED** [review]

**Comment by ImmuniDGR** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true">Messages</th>
    </tr>
  </thead>
  <tbody><tr>
      <td>:book:</td>
      <td>

  :white_check_mark: Commitlint passed
  </td>
    </tr>
  
<tr>
      <td>:book:</td>
      <td>

  :white_check_mark: Mypy (0.770) passed
  </td>
    </tr>
  
<tr>
      <td>:book:</td>
      <td>

  :white_check_mark: Black (19.10b0) passed
  </td>
    </tr>
  
<tr>
      <td>:book:</td>
      <td>Thank you for submitting a pull request! The team will review your submission as soon as possible.</td>
    </tr>
  </tbody>
</table>


<p align="right">
  Generated by :no_entry_sign: <a href="https://danger.systems/js">dangerJS</a> against 40250baeec0adc1c67de88c73f2078664d0c5baf
</p>

**Comment by ImmuniDGR** [comment]
🚔 **Safety Check** 🚔

🔰 **Result** 🔰
- The following files have been modified: yarn.lock, scheduler.py.
- The CircleCI configuration file has been modified.


⚠️ Some configuration files don't match the master branch. If you did not perform these changes, **please rebase on the master branch**.

🛠 **Diagnostic information** 🛠
- CircleCI scheduler pipeline: #33071 (id: f60b83e2-183b-400f-a8dc-96b6b10a37c2)
- Last verified commit: 40250baeec0adc1c67de88c73f2078664d0c5baf
- Time of check: 01/10/2021, 04:49:43 UTC
- The following protected files have been checked for changes: Pipfile, Pipfile.lock, ci/danger/black/config.toml, ci/danger/black/index.ts, ci/danger/commitlint/index.ts, ci/danger/commitlint/rules.ts, ci/danger/danger.ts, ci/danger/mypy/index.ts, ci/danger/mypy/mypy.ini, dangerfile.ts, helpers/\_\_init\_\_.py, helpers/circleci.py, helpers/utils.py, package.json, scheduler.py, scheduler\_config.json, yarn.lock.

**Comment by ImmuniDGR** [comment]
🚔 **Safety Check** 🚔

🔰 **Result** 🔰
<ul>
<li>The following files have been modified: Pipfile.lock, Pipfile, scheduler.py, helpers/utils.py, helpers/circleci.py.</li>
<li>The CircleCI configuration file has been modified.</li>
</ul>

⚠️ Some configuration files don't match the master branch.

🛠 **Diagnostic information** 🛠
- CircleCI scheduler pipeline: #348 (id: 3a8cc7f0-5695-4b93-988d-002cd3f2ebb2)
- Last verified commit: #44140da6cb9a2e6fee38bbf6da07191be1de6344
- Time of check: 28/05/2020, 13:15:45 UTC
- The following protected files have been checked for changes: helpers/__init__.py, package.json, ci/danger/danger.ts, ci/danger/black/index.ts, ci/danger/mypy/index.ts, scheduler_config.json, Pipfile.lock, Pipfile, ci/danger/commitlint/rules.ts, yarn.lock, dangerfile.ts, ci/danger/mypy/mypy.ini, scheduler.py, helpers/utils.py, ci/danger/commitlint/index.ts, ci/danger/black/config.toml, helpers/circleci.py.

**Review by voxsim — APPROVED** [review]

**Comment by ImmuniDGR** [comment]
🚔 **Safety Check** 🚔

🔰 **Result** 🔰
<ul>
<li>The following files have been modified: Pipfile.lock, helpers/utils.py, helpers/circleci.py, Pipfile, scheduler.py.</li>
<li>The CircleCI configuration file has been modified.</li>
</ul>

⚠️ Some configuration files don't match the master branch.

🛠 **Diagnostic information** 🛠
- CircleCI scheduler pipeline: #355 (id: e98191a4-c476-449e-be55-d6dbef966ef3)
- Last verified commit: #40250baeec0adc1c67de88c73f2078664d0c5baf
- Time of check: 28/05/2020, 14:45:45 UTC
- The following protected files have been checked for changes: ci/danger/commitlint/index.ts, scheduler_config.json, yarn.lock, ci/danger/black/index.ts, ci/danger/black/config.toml, dangerfile.ts, ci/danger/mypy/mypy.ini, Pipfile.lock, ci/danger/mypy/index.ts, ci/danger/danger.ts, ci/danger/commitlint/rules.ts, helpers/utils.py, package.json, helpers/__init__.py, helpers/circleci.py, Pipfile, scheduler.py.

**Pull request merged**